src/components/ResourceList/ResourceList.tsx

```diff
@@ -743,9 +761,24 @@ class ResourceList extends React.Component<CombinedProps, State> {
       this.handleSelectMode(true);
     }
 
+    let checkbox: CheckboxHandles | undefined;
+
+    if (isSmallScreen()) {
+      checkbox = checkableButtons.get('bulkSm');
+    } else if (newlySelectedItems.length === 0) {
+      checkbox = checkableButtons.get('plain');
+    } else {
+      checkbox = checkableButtons.get('bulkLg');
+    }
+
     if (onSelectionChange) {
       onSelectionChange(newlySelectedItems);
     }
+
+    // setTimeout ensures execution after the Transition on BulkActions
+    setTimeout(() => {
```

Comment:
cc/ @beefchimi similar to what we were looking at yesterday

Reply:
Similar, except this is able to get away with a `0ms` Timeout 😄 

In this case, is `requestAnimationFrame` an alternative solution the problem? Also... is this needed specifically because of being nested in `react-transition-group`?

Reply:
Yes, it is needed because of `react-transition-group`. They trigger their transition using `setTimeout`. So using `setTimeout` here ensures that theirs is called first. `requestAnimationFrame` was working, but because they are using `setTimeout` we felt it was better here.